lib/Mojolicious.pm

```diff
@@ -175,6 +175,16 @@ sub plugin {
 
 sub server { $_[0]->plugins->emit_hook(before_server_start => @_[1, 0]) }
 
+sub app_starting {
+  my $self = shift;
+
+  # Ensure we run 'before_app_start' only once per process
+  $self->{_started} //= {};
+  return if exists $self->{_started}->{$$};
+  $self->{_started}->{$$} = undef;
+  $self->plugins->emit_hook(before_app_start => $self);
```

Comment:
I find the new name misleading. It implies that it runs before `app->start`.

Reply:
I agree here.  In line with other comment from @s1037989 yesterday.
Naming **is** hard.  What would you choose? `before_app_serving`?  Or perhaps drop the `before_` and say `start`?  Or `init`?

Reply:
`after_server_start`, consistent with `before_server_start`? 😁

Reply:
Sure, but that would mean it has comparable semantics to some "server starting" event and... remember... there is no such thing for CGI yet by your own admission... :grinning: 

That being said, I have absolutely no stake in this naming.  I just want to have this new functionality in **standard** Mojolicious and not just in my own forked framework.  Forking *Open-Source* projects is always... bad.

Reply:
Note that Mojolicious gets forked so much that we had to adopt an official policy for it. https://docs.mojolicious.org/Mojolicious/Guides/Contributing#FORK-POLICY